### PR TITLE
Chapter1/homework/4

### PR DESCRIPTION
**Pull request by kaiomv**
the changes in lesson3tests.cs its because I copied the code from the archive from discord bc-homework, https://github.com/csinn/CSharp-From-Zero-To-Hero/blob/Chapter1/Homework/4/Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs

### CODE REVIEW

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4883521&bid=16180277)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            Console.WriteLine(name + " " + surename + "'s BMI is " + BmiCalc(weight, height));
+
+            Console.ReadLine();
+        }
+        public static float BmiCalc(float weight, float height)
+        {
+            int control = 0;
+            if (height <= 0) control = 1;
+            if (weight <= 0) control += 2;
+            switch (control)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4883521)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            }
+            return name;
+        }
+        public static int PrintAndReadInt(string message)
+        {
+            Console.WriteLine(message);
+            string input = Console.ReadLine();
+            int age;
+            bool isNumber;
+            if (string.IsNullOrEmpty(input)) return 0;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4883521)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+            Console.WriteLine(name + " " + surename + " is " + age + " years old, his weight is " + weight + " kg and his height is " + height + " m ");
+            Console.WriteLine(name + " " + surename + "'s BMI is " + BmiCalc(weight, height));
+
+            Console.ReadLine();
+        }
+        public static float BmiCalc(float weight, float height)
+        {
+            int control = 0;
+            if (height <= 0) control = 1;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4883521)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            if (weight <= 0) control += 2;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4883521)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4883521&bid=16180277)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                return age;
+            }
+        }
+        public static float PrintAndReadFloat(string message)
+        {
+            Console.WriteLine(message);
+            string input = Console.ReadLine();
+            float output;
+            bool isNumber;
+            if (string.IsNullOrEmpty(input)) return 0;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4883521)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4883521&bid=16180277)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+            int control = 0;
+            if (height <= 0) 
+            { 
+                control = 1; 
+            }
+            if (weight <= 0)
+            {
+                control += 2;
+            }
+            switch (control)
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: I love that idea. I also had that one when coding this homework. 👍

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+                    Console.WriteLine("Height cannot be equal or less than zero, but was " + height + ".");
+                    return -1;
+                case (2):
+                    Console.WriteLine("Failed calculating BMI. Reason:");
+                    Console.WriteLine("Weight cannot be equal or less than zero, but was " + weight + ".");
+                    return -1;
+                case (3):
+                    Console.WriteLine("Failed calculating BMI. Reason:");
+                    Console.WriteLine("Weight cannot be equal or less than zero, but was " + weight + ".");
+                    Console.WriteLine("Height cannot be equal or less than zero, but was " + height + ".");
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: Note:
The wording is different on the second output here. Only in case(3)!!

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4883521&bid=16180277)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 8]
```diff
@@ -0,0 +1,126 @@
+﻿using System;
+
+namespace BootCamp.Chapter
+{
+    class Lesson3
+    {
+        public static void Demo()
+        {
+            string name = PrintAndReadString("Hello, what's the name of the person that you want to calculate the BMI?");
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Please write a function to remove duplicate parts in code.

**Comment by Almantask** [comment]
Expired

**Pull request closed without merging**